src/Microsoft.Azure.WebJobs.Host/Triggers/TriggeredFunctionBinding.cs

```diff
@@ -56,7 +58,7 @@ internal class TriggeredFunctionBinding<TTriggerValue> : ITriggeredFunctionBindi
             {
                 ITriggerData triggerData = await _triggerBinding.BindAsync(value, context);
                 triggerProvider = triggerData.ValueProvider;
-                bindingData = triggerData.BindingData;
+                bindingData = triggerData.BindingData.Concat(AppSettingsBinding.CreateBindingData()).ToDictionary(x => x.Key, x => x.Value);
```

Comment:
I think this would be a breaking change for existing applications. E.g. if they moved to the new release and had app settings that happened to coincide with binding data their app is already using (e.g. their own POCO types, etc.).

I'm also wondering if this is a feature that is really useful generally, or if we shouldn't constrain it to just Singleton right now. Singleton is a new feature, so we can't break anyone.